test/unit/lib/plugins/aws/package/compile/functions.test.js

```diff
@@ -2369,22 +1444,71 @@ describe('AwsCompileFunctions', () => {
 describe('lib/plugins/aws/package/compile/functions/index.test.js', () => {
   describe('Provider properties', () => {
     let cfResources;
+    let cfOutputs;
     let naming;
     let serviceConfig;
-    let iamRolePolicyStatements;
 
     before(async () => {
+      const serviceArtifactPath = path.join(process.cwd(), 'testServiceArtifact.zip');
+      await fs.promises.writeFile(serviceArtifactPath, 'packageContent');
+      const functionArtifactPath = path.join(process.cwd(), 'testFunctionArtifact.zip');
+      await fs.promises.writeFile(functionArtifactPath, 'packageContent');
```

Comment:
Instead of creating those files, let's switch the fixture from `function` to `packageArtifact` (it has artifact precreated, and if there's a need we may add an extra one)

_(Ideally fixtures should have all content pre-created, and we should not do additional setup in `before`)_

Reply:
That makes perfect sense :100: I've also added an extra `artifact` so it's possible to test if function can have a different than the one set on service level - please let me know if that approach seems okay to you :bow:

Reply:
Yes, that's totally great!